databricks/koalas/frame.py

```diff
@@ -7851,7 +7864,7 @@ def _reindex_index(self, index, fill_value):
         else:
             joined_df = frame.join(labels, on=index_column, how="right")
 
-        internal = self._internal.with_new_sdf(joined_df)
+        internal = InternalFrame(joined_df, index_map=index_map)
```

Comment:
This removes `column_labels` and doesn't drop `NATURAL_ORDER_COLUMN_NAME`. Also we should update `data_spark_columns`.
Could you follow what's done in `InternalFrame. with_new_sdf`.

Reply:
Yes, I am working on it